lib/collection/src/collection_manager/segments_updater.rs

```diff
@@ -39,6 +39,16 @@ pub(crate) fn delete_points(
     Ok(res)
 }
 
+/// Tries to delete points from all segments, returns number of actually deleted points
+pub(crate) fn delete_points_by_filter(
+    segments: &SegmentHolder,
+    op_num: SeqNumberType,
+    filter: &Filter,
+) -> CollectionResult<usize> {
+    let point_ids = segments.search_points(op_num, filter)?;
+    Ok(delete_points(segments, op_num, &point_ids)?)
```

Comment:
makes total sense! Thanks for the hint! However it's not possible anymore to return the number of deleted points,because the `apply_segments` function returns an `Result<usize, _?>` where usize is the number of processed segments. One could log things, in general, but thats which is reflected in the API anymore.

Reply:
One solution could be to accumulate number of deleted points inside the callback function of `apply_segments`

Reply:
ok, this would mean that the function would return `OperationResult<usize>` instead of now `OperationResult<bool>`. This is not easily possible, since `apply_segments` is also used with `create_field_index` and `delete_field_index` which both return `OperationResult<bool>` with a semantically intended reason. I could introduce an enum as return type:

```rust
pub enum AppliedSegmentResult {
  Applied(bool),
  AffectedPoints(usize)
}
```

and handle this in apply segments respectivley to return the respective values to the callers of that function.

Reply:
example implementation attached...

Reply:
That is not actually necessary, check out my suggestion below